File: sources/codes/frontend/cp-trip-sharing-frontend/src/app/model/PostFilter.ts
export class PostFilter {
    // timePeriod is in [today, this_week, this_month, this_year, all_time]
    timePeriod: string;

    // topics is list of topicId
    topics: string[];

    // search by text
    search: string;

    // locationId of location in google map
    locationId: string;
}

File: sources/codes/backend/cp-trip-sharing-backend/src/Services/PostService/PostService/Models/PostFilter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace PostService.Models
{
    public class PostFilter
    {
        public string Search { get; set; }

        public string LocationId { get; set; }

        public string TimePeriod { get; set; }

        public List<string> Topics { get; set; }
    }
}
